javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -204,6 +204,14 @@ public Map<Request, List<ProxyMethod>> map(Operation operation) {
             List<ProxyMethodParameter> parameters = new ArrayList<>();
             List<ProxyMethodParameter> allParameters = new ArrayList<>();
             List<ProxyMethod> proxyMethods = new ArrayList<>();
+            // add content-type parameter to allParameters when body is optional and there is single content type
+            boolean isBodyParamRequired = request.getParameters()
+                    .stream().filter(p -> p.getProtocol() != null && p.getProtocol().getHttp() != null ? p.getProtocol().getHttp().getIn() == RequestParameterLocation.BODY : false)
+                    .map(Parameter::isRequired).findFirst().orElse(false);
```

Comment:
nit, move this (and the `MethodUtil.getContentTypeCount(operation.getRequests()) == 1` check) after the if block of `settings.isDataPlaneClient()`.

reason, if some variable only useful for DPG, avoid it exposed to code not really need to see that var).